src/test/java/com/google/api/gax/grpc/CancellationTest.java

```diff
@@ -0,0 +1,298 @@
+
+    ImmutableSet<Status.Code> retryable = ImmutableSet.<Status.Code>of(Status.Code.UNAVAILABLE);
+    UnaryApiCallable<Integer, Integer> callable =
+        UnaryApiCallable.<Integer, Integer>create(callInt)
+            .retryableOn(retryable)
+            .retrying(FAST_RETRY_SETTINGS, executor, fakeClock);
+
+    ListenableFuture<Integer> resultFuture = callable.futureCall(0);
+    resultFuture.cancel(true);
+    resultFuture.get();
+  }
+
+  private static class LatchCountDownScheduler implements UnaryApiCallable.Scheduler {
+    private final ScheduledExecutorService executor;
+    private final CountDownLatch latch;
+
+    LatchCountDownScheduler(CountDownLatch latch) {
+      this.executor = new ScheduledThreadPoolExecutor(1);
+      this.latch = latch;
+    }
+
+    @Override
+    public ScheduledFuture<?> schedule(Runnable runnable, long delay, TimeUnit unit) {
+      latch.countDown();
+      return executor.schedule(runnable, delay, unit);
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+      return executor.shutdownNow();
+    }
+  }
+
+  private void cancelAfterLatchCountDown(
+      final ListenableFuture<Integer> resultFuture, final CountDownLatch latch) {
+    Thread t =
+        new Thread(
+            new Runnable() {
+              @Override
+              public void run() {
+                try {
+                  latch.await();
+                } catch (InterruptedException e) {
+                  Thread.currentThread().interrupt();
+                }
+                resultFuture.cancel(true);
+              }
+            });
+    t.start();
+  }
+
+  private static class CancellationTrackingFuture<RespT> extends AbstractFuture<RespT> {
+    private volatile boolean cancelled = false;
+
+    public static <RespT> CancellationTrackingFuture<RespT> create() {
+      return new CancellationTrackingFuture<>();
+    }
+
+    private CancellationTrackingFuture() {}
+
+    @Override
+    protected void interruptTask() {
+      cancelled = true;
+    }
+
+    public boolean isCancelled() {
+      return cancelled;
+    }
+  }
+
+  private static class LatchCountDownFutureCallable<RequestT, ResponseT>
+      implements FutureCallable<RequestT, ResponseT> {
+    private CountDownLatch callLatch;
+    private List<ListenableFuture<ResponseT>> injectedFutures;
+
+    @SuppressWarnings("unchecked")
+    public LatchCountDownFutureCallable(
+        CountDownLatch callLatch, ListenableFuture<ResponseT> injectedFuture) {
+      this(callLatch, Lists.newArrayList(injectedFuture));
+    }
+
+    public LatchCountDownFutureCallable(
+        CountDownLatch callLatch, List<ListenableFuture<ResponseT>> injectedFutures) {
+      this.callLatch = callLatch;
+      this.injectedFutures = Lists.newArrayList(injectedFutures);
+    }
+
+    @Override
+    public ListenableFuture<ResponseT> futureCall(RequestT request, CallContext context) {
+      callLatch.countDown();
+      if (injectedFutures.size() == 1) {
```

Comment:
It's not required - I have removed it for simplicity. It must be a carryover from when I was writing the tests.